automation/orchestrator.go

```diff
@@ -0,0 +1,165 @@
+package automation
+
+import (
+	"context"
+	"fmt"
+
+	golog "github.com/ipfs/go-log"
+	"github.com/qri-io/ioes"
+	"github.com/qri-io/qri/automation/run"
+	"github.com/qri-io/qri/automation/workflow"
+	"github.com/qri-io/qri/event"
+	"github.com/qri-io/qri/profile"
+)
+
+var (
+	log = golog.Logger("automation")
+)
+
+// Options provides details to NewOrchestrator
+// NewOrchestrator will alter Options by applying any provied Option functions
+type Options struct {
+	workflowStore workflow.Store
+	saveFactory   SaveFactory
+	applyFactory  ApplyFactory
+}
+
+// Option is a function that manipulates config details when fed to NewOrchestrator. Fields on the opts parameter may be null
+type Option func(opts *Options) error
+
+// OptWorkflowStore allows you to pass in a custom workstore to the Orchestrator
+func OptWorkflowStore(ws workflow.Store) Option {
+	return func(opts *Options) error {
+		opts.workflowStore = ws
+		return nil
+	}
+}
+
+// Save is what the Orchestrator will call to run a transform in during a workflow
+type Save func(ctx context.Context, streams ioes.IOStreams, w *workflow.State, runID string) error
```

Comment:
Thing I've been considering: should I change `Save` to `Run` and `Apply` to `DryRun`, since that is the language we use in the rest of this package?

Reply:
I'm a fan changing `Save` to `Run`. I think it solves the confusion both of us encountered with this function, expecting save to get a `*dataset.Dataset` somewhere.

I'd like to avoid renaming `Apply` to `DryRun`. @dustmop's [RFC](https://github.com/qri-io/rfcs/blob/master/text/0027-transform_application.md) formalized our switch to using `Apply`.